elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTransport.java

```diff
@@ -224,6 +224,16 @@ void closeScroll(Iterable<String> scrollIds) {
       return rawHttp(ElasticsearchJson.Result.class).apply(post);
     };
   }
+  
+  
+
+  @Override
+  protected void finalize() throws Throwable {
```

Comment:
Finalizers are a java concept that is strongly discouraged for quite some time now. Moreover, they are going to [disappear soon](https://www.youtube.com/watch?v=eDgBnjOid-g) so let's find another way to handle this.

Reply:
[JEP 421: Deprecate Finalization for Removal](https://openjdk.java.net/jeps/421?continueFlag=abef66f10493295d8376160baf41892b) This address have some way to replace the finalize method and Why we should avoid using it. I hope it can help.